### PR TITLE
Small Tweak

### PR DESCRIPTION
**Pull request by time-killer-games**
yes, *another* widget tweak...

### CODE REVIEW

**Comment by time-killer-games** [comment]
@JoshDreamland ready for review

File: ENIGMAsystem/SHELL/Widget_Systems/xlib/dialogs.cpp [hunk 1]
```diff
@@ -18,17 +18,28 @@
...
+  Display *d = XOpenDisplay(NULL);
+  Atom aKWinRunning = XInternAtom(d, "KWIN_RUNNING", True);
+  bool bKWinRunning = (aKWinRunning != None);
+  XCloseDisplay(d);
+  return bKWinRunning;
+}
+
 bool widget_system_initialize() {
+  // Defaults to the GUI toolkit (GTK+/Qt) that matches Desktop Environment.
+  current_widget_engine = kwin_running() ? kdialog_widgets : zenity_widgets;
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Widget_Systems/xlib/dialogs.cpp]: KDE running doesn't really guarantee that kdialog is available. If both Zenity and KDialog are available, then you should of course prefer whichever system is running, but odds are, the user has neither one installed. And if they do, they almost certainly only have one or the other. That said, this is a better guess than what we had before.

**Comment by time-killer-games** [comment]
@JoshDreamland u mudge? This will make hugar very happy

**Pull request merged**